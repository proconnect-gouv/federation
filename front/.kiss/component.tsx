import React from 'react';

interface DumbComponentProps {
  className: string;
}

export const DumbComponent: React.FC<DumbComponentProps> = React.memo(
  ({ className }: DumbComponentProps) => {
    return <div />;
  },
);

DumbComponent.displayName = 'DumbComponent';
