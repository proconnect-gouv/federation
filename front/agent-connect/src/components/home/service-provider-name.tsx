/* istanbul ignore file */
import React from 'react';

const ServiceProviderNameComponent = (): JSX.Element => {
  return (
    <section className="row mt-5 mb-8 text-center">
      <p className="h4 col-12">Je choisis un compte pour me connecter sur</p>
      <h1 className="col-12 text-center font-weight-bold text-primary">
        <span>Name of the fournisseur</span>
      </h1>
    </section>
  );
};

export default ServiceProviderNameComponent;
