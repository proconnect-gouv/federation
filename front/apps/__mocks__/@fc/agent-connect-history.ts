export const useUserHistory = jest.fn(() => [
  { active: false, display: true, name: 'mock-name-1', uid: 'mokc-uid-1' },
  { active: false, display: true, name: 'mock-name-2', uid: 'mokc-uid-2' },
  { active: false, display: true, name: 'mock-name-3', uid: 'mokc-uid-3' },
  { active: false, display: true, name: 'mock-name-4', uid: 'mokc-uid-4' },
]);

export const useAddToUserHistory = jest.fn(() => jest.fn());

export const useRemoveFromUserHistory = jest.fn(() => jest.fn());
