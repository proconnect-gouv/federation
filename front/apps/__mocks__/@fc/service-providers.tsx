export const ServiceProviderItemComponent = jest.fn(({ datapassId }) => (
  <div>ServiceProviderItemComponent {datapassId}</div>
));

export {
  ServiceProviderEditActionTypes,
  ServiceProvidersActionTypes,
  ServiceProvidersPermissionTypes,
  ServiceProviderViewActionTypes,
} from '@fc/service-providers';
