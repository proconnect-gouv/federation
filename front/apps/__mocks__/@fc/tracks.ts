export const TracksListComponent = jest.fn();
