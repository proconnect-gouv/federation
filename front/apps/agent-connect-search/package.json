{
  "name": "@fc/agent-connect-search",
  "version": "0.1.0",
  "private": true,
  "main": "src/index.ts",
  "dependencies": {
    "@fc/common": "^0.1.0",
    "@fc/routing": "^0.1.0",
    "fuse.js": "^6.5.0",
    "lodash.debounce": "^4.0.8",
    "react": "^18.2.0"
  }
}
