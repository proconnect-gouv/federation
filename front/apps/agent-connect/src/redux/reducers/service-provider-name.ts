import { ACTION_TYPES } from '../../constants';

export const serviceProviderName = (state = '', action: any) => {
  switch (action.type) {
    case ACTION_TYPES.MINISTRY_LIST_LOAD_START:
      return '';
    case ACTION_TYPES.MINISTRY_LIST_LOAD_COMPLETED:
      return action?.payload?.serviceProviderName || state;
    default:
      return state;
  }
};
