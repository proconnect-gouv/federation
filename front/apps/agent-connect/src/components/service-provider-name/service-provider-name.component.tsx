import classNames from 'classnames';
import React, { useContext } from 'react';
import { useMediaQuery } from 'react-responsive';

import { AgentConnectSearchContext } from '@fc/agent-connect-search';

export const ServiceProviderNameComponent = React.memo(() => {
  const gtTablet = useMediaQuery({ query: '(min-width: 768px)' });
  const { payload } = useContext(AgentConnectSearchContext);
  const { serviceProviderName } = payload;

  return (
    // Class CSS
    // eslint-disable-next-line @typescript-eslint/naming-convention
    <section className={classNames({ 'text-center': gtTablet })} data-testid="wrapper">
      <h1
        // Class CSS
        // eslint-disable-next-line @typescript-eslint/naming-convention
        className={classNames('mx16 mb16', { 'fr-text-lg': !gtTablet, fs28: gtTablet })}
        data-testid="title">
        Je choisis un compte pour me connecter sur
      </h1>
      <h2
        className={classNames('is-extra-bold is-blue-agentconnect mx16', {
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fs32 mb40': !gtTablet,
          // Class CSS
          // eslint-disable-next-line @typescript-eslint/naming-convention
          'fs40 mb64': gtTablet,
        })}
        data-testid="subtitle">
        <span>{serviceProviderName}</span>
      </h2>
    </section>
  );
});

ServiceProviderNameComponent.displayName = 'ServiceProviderNameComponent';
