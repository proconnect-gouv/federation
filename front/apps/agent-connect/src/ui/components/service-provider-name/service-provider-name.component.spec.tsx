import { render } from '@testing-library/react';
import { mocked } from 'jest-mock';
import { useMediaQuery } from 'react-responsive';

import { AgentConnectSearchContext } from '@fc/agent-connect-search';

import { ServiceProviderNameComponent } from './service-provider-name.component';

describe('ServiceProviderNameComponent', () => {
  const agentConnectSearchContextMock = {
    hasSearched: false,
    payload: {
      csrfToken: 'any-csrf-token-mock',
      identityProviders: [],
      isLoaded: false,
      ministries: [],
      serviceProviderName: 'Service provider name mock',
    },
    searchResults: [],
    setSearchTerm: jest.fn(),
  };

  it('should have call useMediaQuery with tablet responsive query', () => {
    // when
    render(<ServiceProviderNameComponent />);
    // then
    expect(useMediaQuery).toHaveBeenNthCalledWith(1, { query: '(min-width: 768px)' });
  });

  it('should have a section title', () => {
    // then
    const { getByText } = render(<ServiceProviderNameComponent />);
    const element = getByText('Je choisis un compte pour me connecter sur');
    // when
    expect(element).toBeInTheDocument();
  });

  it('should have a serviceProviderName label', () => {
    // then
    const { getByText } = render(
      <AgentConnectSearchContext.Provider value={agentConnectSearchContextMock}>
        <ServiceProviderNameComponent />
      </AgentConnectSearchContext.Provider>,
    );
    const element = getByText('Service provider name mock');
    // when
    expect(element).toBeInTheDocument();
  });

  it('should match the snapshot for a desktop viewport', () => {
    // when
    const { container } = render(<ServiceProviderNameComponent />);
    // then
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot for a tablet viewport', () => {
    // given
    mocked(useMediaQuery).mockReturnValueOnce(false);
    // when
    const { container } = render(<ServiceProviderNameComponent />);
    // then
    expect(container).toMatchSnapshot();
  });

  it('should have called useMediaQuery with tablet responsive query', () => {
    // then
    render(<ServiceProviderNameComponent />);
    // when
    expect(useMediaQuery).toHaveBeenNthCalledWith(1, { query: '(min-width: 768px)' });
  });
});
