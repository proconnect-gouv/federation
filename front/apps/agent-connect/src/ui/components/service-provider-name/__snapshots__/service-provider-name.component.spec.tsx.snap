// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceProviderNameComponent should match the snapshot for a desktop viewport 1`] = `
<div>
  <section
    class="fr-px-2w text-center"
    data-testid="wrapper"
  >
    <h1
      class="fr-h4 fr-mb-2w"
      data-testid="title"
    >
      Je choisis un compte pour me connecter sur
    </h1>
    <h2
      class="fr-h1 fr-mb-8w"
      data-testid="subtitle"
    >
      <span />
    </h2>
  </section>
</div>
`;

exports[`ServiceProviderNameComponent should match the snapshot for a tablet viewport 1`] = `
<div>
  <section
    class="fr-px-2w"
    data-testid="wrapper"
  >
    <h1
      class="fr-h4 fr-mb-2w"
      data-testid="title"
    >
      Je choisis un compte pour me connecter sur
    </h1>
    <h2
      class="fr-h1 fr-mb-5w"
      data-testid="subtitle"
    >
      <span />
    </h2>
  </section>
</div>
`;
